UI/EncyclopediaDetailPanel.cpp

```diff
@@ -2428,7 +2428,27 @@ namespace {
         // show image of planet environment at the top of the suitability report
         std::string planet_type = boost::lexical_cast<std::string>(planet->Type());
         boost::algorithm::to_lower(planet_type);
-        detailed_description += "<img src=\"encyclopedia/planet_environments/" + planet_type + ".png\"></img>";
+
+        namespace fs = boost::filesystem;
+        std::string filename;
+        int matches = 0;
+        fs::directory_iterator end_it;
+        for (fs::directory_iterator it(ClientUI::ArtDir() / "encyclopedia" / "planet_environments"); it != end_it; ++it) {
+            try {
+                if (fs::exists(*it) && !fs::is_directory(*it) && boost::algorithm::starts_with(it->path().filename().string(), planet_type)) {
```

Comment:
is this safe for Windows wide character filenames?

Reply:
This is pretty much the same code as [PrefixedTexturesAndDist](https://github.com/freeorion/freeorion/blob/d8164191fb4a800a0ed206393ebb300f36d29ba0/UI/ClientUI.cpp#L1121), so if that works, this should work too.

Note that it's only comparing the filename portion of the path, so unless there's a planet type with non-ASCII in its name, the problem should never even come up (it doesn't matter if the directory is located in a user directory with Cyrillic characters, for example).